src/main/kotlin/ch/unibas/dmi/dbis/cottontail/server/grpc/helper/DataExtensions.kt

```diff
@@ -469,39 +489,19 @@ fun CottontailGrpc.Vector.toBooleanVectorValue(): BooleanVectorValue = when (thi
-    CottontailGrpc.Data.DataCase.COMPLEXDATA -> ComplexVectorValue(Array(1) { Complex(floatArrayOf(this.complexData.real, this.complexData.imaginary)) }) // TODO { this.complexData })
-    CottontailGrpc.Data.DataCase.VECTORDATA -> this.vectorData.toComplexVectorValue()
-    CottontailGrpc.Data.DataCase.NULLDATA -> null
-    CottontailGrpc.Data.DataCase.DATA_NOT_SET -> throw QueryException.UnsupportedCastException("A value of NULL cannot be cast to VECTOR[COMPLEX].")
-    null -> throw QueryException.UnsupportedCastException("A value of NULL cannot be cast to VECTOR[COMPLEX].")
-}
-
 /**
  * Returns the value of [CottontailGrpc.Vector] as [ComplexVectorValue].
  *
  * @return [ComplexVectorValue] values
  * @throws QueryException.UnsupportedCastException If cast is not possible.
  */
 fun CottontailGrpc.Vector.toComplexVectorValue(): ComplexVectorValue = when (this.vectorDataCase) {
-    CottontailGrpc.Vector.VectorDataCase.DOUBLEVECTOR -> throw QueryException.UnsupportedCastException("A value of DOUBLE cannot be cast to VECTOR[COMPLEX].")
-    CottontailGrpc.Vector.VectorDataCase.FLOATVECTOR -> throw QueryException.UnsupportedCastException("A value of FLOAT cannot be cast to VECTOR[COMPLEX].")
-    CottontailGrpc.Vector.VectorDataCase.LONGVECTOR -> throw QueryException.UnsupportedCastException("A value of LONG cannot be cast to VECTOR[COMPLEX].")
-    CottontailGrpc.Vector.VectorDataCase.INTVECTOR -> throw QueryException.UnsupportedCastException("A value of INT cannot be cast to VECTOR[COMPLEX].")
+    CottontailGrpc.Vector.VectorDataCase.DOUBLEVECTOR -> ComplexVectorValue(Array(this.complexVector.vectorList.size) { i -> Complex(floatArrayOf(this.complexVector.vectorList[i].real, 0.0f)) })
+    CottontailGrpc.Vector.VectorDataCase.FLOATVECTOR -> ComplexVectorValue(Array(this.complexVector.vectorList.size) { i -> Complex(floatArrayOf(this.complexVector.vectorList[i].real, 0.0f)) })
+    CottontailGrpc.Vector.VectorDataCase.LONGVECTOR -> ComplexVectorValue(Array(this.complexVector.vectorList.size) { i -> Complex(floatArrayOf(this.complexVector.vectorList[i].real, 0.0f)) })
+    CottontailGrpc.Vector.VectorDataCase.INTVECTOR -> ComplexVectorValue(Array(this.complexVector.vectorList.size) { i -> Complex(floatArrayOf(this.complexVector.vectorList[i].real, 0.0f)) })
     CottontailGrpc.Vector.VectorDataCase.BOOLVECTOR -> throw QueryException.UnsupportedCastException("A value of BOOL cannot be cast to VECTOR[COMPLEX].")
-    CottontailGrpc.Vector.VectorDataCase.COMPLEXVECTOR -> ComplexVectorValue(Array(this.complexVector.vectorList.size) { Complex(floatArrayOf(this.complexVector.vectorList[0].real, this.complexVector.vectorList[0].imaginary)) }) // TODO ComplexVectorValue(this.complexVector.vectorList)
+    CottontailGrpc.Vector.VectorDataCase.COMPLEXVECTOR -> ComplexVectorValue(Array(this.complexVector.vectorList.size) { i -> Complex(floatArrayOf(this.complexVector.vectorList[i].real, this.complexVector.vectorList[i].imaginary)) })
```

Comment:
Something long the line of 

```kotlin

ComplexVectorValue(this.complexVector.vectorList.map { Complex(floatArrayOf(it.real, it.imaginary)) }.toTypedArray()

```

would be more concise and more the "Kotlin-way" of doing things. Please double-check syntax, since I don't have a compiler up and running.